src/main/java/com/nextcloud/client/integration/deck/DeckApi.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Nextcloud application
+ *
+ * @author Stefan Niedermann
+ * Copyright (C) 2020 Stefan Niedermann <info@niedermann.it>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.nextcloud.client.integration.deck;
+
+import android.app.PendingIntent;
+
+import com.nextcloud.client.account.User;
+import com.nextcloud.java.util.Optional;
+import com.owncloud.android.lib.resources.notifications.models.Notification;
+
+import androidx.annotation.NonNull;
+
+/**
+ * This API is for an integration with the <a href="https://github.com/stefan-niedermann/nextcloud-deck">Nextcloud
+ * Deck</a> app for android.
+ */
+public interface DeckApi {
+
+    /**
+     * Creates a PendingIntent that can be used in a NotificationBuilder to start the Deck app
+     *
+     * @param notification Notification object that should be processed
+     * @param user         The user that is affected by the notification
+     * @return Optional with a PendingIntent or an empty Optional if the notification is not from the
+     * <a href="https://apps.nextcloud.com/apps/deck">Deck server app</a>.
+     */
```

Comment:
The diagram tells that it passes account and some kind of "event".


Can we clarify what this API is doing exactly?

My posposal (after consulting the source code):

```suggestion
    /**
    * Creates a PendingIntent that can be used in a NotificationBuilder to open the notification link in Deck app 
    * @see Deck Server App - https://apps.nextcloud.com/apps/deck
    *
    * @param notification Notification that could be forwarded to Deck
    * @param user         The user that is affected by the notification
    * @return If notification can be consumed by Deck, a PendingIntent openinig notification link in Deck app; empty value otherwise
    */
```

Reply:
I don't want to block this PR - we can follow-up when merging #5892

Reply:
I adapted your proposal (but kept the `<a href...`-style for the `@see`-annotation ([because it will then be rendered as a link](https://stackoverflow.com/a/10683345))

> The diagram tells that it passes account and some kind of "event".

This can currently be for example

- User A assigns a card to User B
- User A mentions User B in a comment on a card
- ...

But this are internals of the Deck app and can change. The actual payload / content of the event is not relevant for the File app and should therefore omitted in the documentation IMHO.

Reply:
> But this are internals of the Deck app and can change. The actual payload / content of the event is not relevant for the File app

Agree, if Deck content is opaque, it's implementation detail that should not be leaking into the caller.